linker/shared/src/main/scala/org/scalajs/linker/frontend/modulesplitter/MaxModuleAnalyzer.scala

```diff
@@ -37,66 +40,24 @@ private[modulesplitter] final class MaxModuleAnalyzer extends ModuleAnalyzer {
       // Fast path.
       new SingleModuleAnalysis(info.publicModuleDependencies.head._1)
     } else {
-      val allTags = new Tagger(info).tagAll()
-
-      val moduleIDs =
-        buildModuleIDs(info.publicModuleDependencies.keys, allTags.flatMap(_._2._2))
-
-      val moduleMap = allTags.map {
-        case (className, tags) => className -> moduleIDs(tags)
-      }.toMap
+      val prefix = internalModuleIDPrefix(info.publicModuleDependencies.keys)
+      val moduleMap = new Tagger(info).tagAll(prefix)
 
       new FullAnalysis(moduleMap)
     }
   }
 
-  private def buildModuleIDs(publicIDsUnordered: Iterable[ModuleID],
-      dynamicIDsUnordered: Iterable[ClassName]): Map[(Set[ModuleID], Set[ClassName]), ModuleID] = {
-    /* We build the new module IDs independent of the actually present
-     * modules to ensure stability.
-     *
-     * We sort the ModuleIDs to not depend on map iteration order (or the
-     * order of the input files).
-     */
-    val publicIDs = sortedSet(publicIDsUnordered)(Ordering.by[ModuleID, String](_.id))
-    val dynamicIDs = sortedSet(dynamicIDsUnordered)
-
-    /* Internal ModuleIDs are simply generated with consecutive numbers.
-     * (earlier versions concatenated all the tags; which lead to too
-     * long names, see #4499).
-     */
-    val internalIDs = Iterator.from(0)
-      .map(i => ModuleID(s"internal-$i"))
-      .filterNot(publicIDs.contains(_)) // avoid module ID collisions.
-
-    val tups = for {
-      dynamicModules <- dynamicIDs.subsets()
-      publicModules <- publicIDs.subsets()
-      if publicModules.nonEmpty || dynamicModules.nonEmpty
-    } yield {
-      val moduleID = {
-        if (dynamicModules.isEmpty && publicModules.size == 1) {
-          /* Classes tagged with exactly a single public module, get assigned to
-           * that public module.
-           */
-          publicModules.head
-        } else {
-          // for all other tag combinations, we generate a new internal module
-          internalIDs.next()
-        }
-      }
-
-      (publicModules, dynamicModules) -> moduleID
-    }
-
-    tups.toMap
-  }
-
-  private def sortedSet[T: Ordering](s: Iterable[T]): immutable.SortedSet[T] = {
-    // Best way I could find to create a SortedSet from an Iterable (i.e. not Seq) :-/
-    val b = immutable.SortedSet.newBuilder[T]
-    s.foreach(b += _)
-    b.result()
+  /** Create a prefix for internal modules.
+   *
+   *  Chosen such that it is not a prefix of any public module ID.
+   *  This ensures that a generated internal module ID never collides with a
+   *  public module ID.
+   */
+  private def internalModuleIDPrefix(publicIDs: Iterable[ModuleID]): String = {
+    Iterator
+      .iterate("internal$")(_ + "$")
```

Comment:
Consider using `-` instead of `$` to create more "idiomatic" Internet file names.
```suggestion
      .iterate("internal-")(_ + "-")
```